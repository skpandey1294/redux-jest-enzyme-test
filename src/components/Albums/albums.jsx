import React, { Component } from 'react'

import '../../assets/styles/Albums.css'

import Typography from '@material-ui/core/Typography'

import { Link } from 'react-router-dom'
import { fetchAlbums } from '../../redux/async-api/albums.js'
import { connect } from 'react-redux'

class Albums extends Component {
  componentDidMount() {
    this.props.albumsList()
  }

  render() {
    const { loading, albums, error } = this.props

    const albumsList = albums.map((album) => {
      return (
        <Link key={`album-${album.id}`} to={`/album/${album.id}`}>
          <div id={`album-${album.id}`} className="card">
            <Typography gutterBottom component="p">
              <b>Title</b>:{album.title}
            </Typography>
          </div>
        </Link>
      )
    })
    return loading === true ? <div>Loading...</div> : ( error !== '' ? <div>Something went wrong</div> :
      <div className="container1">{albumsList}</div> )
    
  }
}

const mapStateToProps = (state) => {
  return {
    loading: state.albumsReducer.loading,
    albums: state.albumsReducer.albums,
    error: state.albumsReducer.error,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    albumsList: () => dispatch(fetchAlbums()),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Albums)
