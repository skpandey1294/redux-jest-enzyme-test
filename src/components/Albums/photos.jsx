import React, { Component } from 'react'

import { fetchAlbumPhotos } from '../../redux/async-api/albumPhotos.js'
import { connect } from 'react-redux'

import Slider from './slider'

class Photos extends Component {

    componentDidMount() {
        const { albumPhotos } = this.props
        const { albumId } = this.props.match.params

        albumPhotos(albumId)
    }

    render() {

        const { loading, photos, error } = this.props

        return (
        loading === true ? <div>Loading...</div> : ( error === '' ? <Slider images={photos}/> : <div>{error.message}</div> )
        )
    }
}

const mapStateToProps = (state) => {
  return {
    loading: state.albumPhotosReducer.loading,
    photos: state.albumPhotosReducer.photos,
    error: state.albumPhotosReducer.error
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    albumPhotos: (albumId) => dispatch(fetchAlbumPhotos(albumId)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Photos)