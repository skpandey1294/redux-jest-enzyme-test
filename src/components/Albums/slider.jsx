import React from 'react'
import '../../assets/styles/Slider.css'

import { keyframes } from 'styled-components'

function Slider({ images }){


    const length = 100/images.length

    let totalTansition = images.map((image, index) => {
        return`${length * index}%  { transform: translateX(-${100*index}%)}` })

    let slideShow = keyframes`
    ${totalTansition}
    `

        let slider = images.map((image, index) => {
            return (
                <div key={index} className="slide" style={{ animation: `${slideShow} ${images.length * 4}s ease-out 1s infinite`}}>
                    <img className="img" src={image.url} alt="slide-img"></img>
                </div>
            )
        })


        return (
            <div className="slider">
                {slider}
            </div>
        )
    }
export default Slider;
