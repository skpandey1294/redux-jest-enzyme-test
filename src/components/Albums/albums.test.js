import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import Albums from './albums';

const mockStore = configureMockStore();
const store = mockStore({});

let wrapped = shallow(
    <Provider store={store}>
        <Albums />
    </Provider>
)


describe("Albums Component", () => {

    it("Albums should render without throwing an error", () => {
        expect(wrapped.debug()).toMatchSnapshot();
    });
    
});