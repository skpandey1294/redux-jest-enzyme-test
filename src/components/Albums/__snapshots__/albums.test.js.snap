// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums Component Albums should render without throwing an error 1`] = `
"<ContextProvider value={{...}}>
  <Component />
</ContextProvider>"
`;
